### PR TITLE
Revamp MLP UI Sidebar Menu

### PR DESCRIPTION
**Pull request by tiopramayudi**
Change the looks and experience of sidebar menu. In this PR only UI files that modified. Below are attached screenshots

Screen Shot 2021-03-31 at 11 01 14

When merlin integrate with new sidebar menu, merlin app is highlighted because currently active app is merlin
Screen Shot 2021-03-31 at 11 08 54

Sidebar menu will be moved to Header, hence to use this new sidebar menu, user just need to specify the header
```
<Header
            appIcon="graphApp"
            docLinks={config.DOC_LINKS}
 />
```
where docLinks is list of documentation that relevant to be shown which has format like:
```
[
   {
      "href":"https://github.com/gojek/merlin/blob/main/docs/getting-started/README.md",
      "label":"Merlin User Guide"
   },
   {
      "href":"https://github.com/gojek/turing",
      "label":"Turing User Guide"
   },
   {
      "href":"https://docs.feast.dev/user-guide/overview",
      "label":"Feast User Guide"
   }
]
```

### CODE REVIEW

File: ui/packages/lib/src/components/nav_drawer/NavDrawer.js [hunk 1]
```diff
@@ -52,29 +39,123 @@ export const NavDrawer = ({ homeUrl = "/", appLinks }) => {
     }
   ];
 
+  const docLinks = [
```
Comment (pradithya) [inline ui/packages/lib/src/components/nav_drawer/NavDrawer.js:42]: This has to be configurable and I don't think we should include link to gojek internal documentations

File: ui/packages/app/src/PrivateLayout.js [hunk 2]
```diff
@@ -20,10 +18,9 @@ export const PrivateLayout = Component => {
               navigate(`/projects/${projectId}/settings`)
             }
           />
-          <CurrentProjectContext.Consumer>
-            {({ projectId }) => projectId && <NavDrawer />}
-          </CurrentProjectContext.Consumer>
-          <Component {...props} />
+          <div style={{ paddingTop: "49px" }}>
```
Comment (pradithya) [inline ui/packages/app/src/PrivateLayout.js:21]: Can we not do in-line style?

File: ui/packages/lib/src/components/nav_drawer/NavDrawer.js [hunk 3]
```diff
@@ -52,29 +39,123 @@ export const NavDrawer = ({ homeUrl = "/", appLinks }) => {
...
+  const [docsExpanded, toggleDocsExpanded] = useToggle(true);
   return (
-    <EuiNavDrawer>
+    <EuiCollapsibleNav
+      aria-label="Main navigation"
+      isOpen={navIsOpen}
+      showCloseButton={false}
+      isDocked={navIsDocked}
+      showButtonIfDocked={true}
+      style={{ top: "49px", height: "calc(100% - 49px)", width: "262px" }}
```
Comment (pradithya) [inline ui/packages/lib/src/components/nav_drawer/NavDrawer.js:75]: Can we remove in-line stype?

File: ui/packages/lib/src/components/nav_drawer/NavDrawer.js [hunk 4]
```diff
@@ -52,29 +39,123 @@ export const NavDrawer = ({ homeUrl = "/", appLinks }) => {
...
+              listItems={docLinks}
+              maxWidth="none"
+              color="subdued"
+              gutterSize="s"
+              size="s"
+            />
+          </EuiCollapsibleNavGroup>
         </EuiFlexItem>
+
+        <EuiHorizontalRule margin="none" />
```
Comment (ariefrahmansyah) [inline ui/packages/lib/src/components/nav_drawer/NavDrawer.js:128]: Please move this rule after the Project Setting button. Similar to the previous sidebar:
image

File: ui/packages/lib/src/components/nav_drawer/NavDrawer.js (hunk 1 above)
Comment (ariefrahmansyah) [reply ui/packages/lib/src/components/nav_drawer/NavDrawer.js:42]: +1 on making it configurable.

Comment (tiopramayudi) [reply ui/packages/lib/src/components/nav_drawer/NavDrawer.js:42]: 🤦 just realised this. Should we make it configurable from db?

**Comment by romanwozniak** [comment]
I'm not sure if it's a good idea to have component-level user manuals be shown at once, independently from what product is currently in-use.

**Comment by romanwozniak** [comment]
Since we already touching sidebar, I think it makes sense to make product menu items (e.g. 'Merlin', 'Feast') to be expandable and have quick-access links to specific resources of a relevant component under them.

Somethin like:

v Merlin
   – Models
   – Endpoints
   – Batch Jobs
v Feast
   – Entities
   – Feature Sets
...

**Review by pradithya — APPROVED** [review]
LGTM, Can you explicitly mention in the MR description how the new NavDrawer behave especially in term of how respective product can add their docs link.

**Comment by tiopramayudi** [comment]
> Since we already touching sidebar, I think it makes sense to make product menu items (e.g. 'Merlin', 'Feast') to be expandable and have quick-access links to specific resources of a relevant component under them.
> 
> Somethin like:
> 
> v Merlin
> – Models
> – Endpoints
> – Batch Jobs
> v Feast
> – Entities
> – Feature Sets
> ...

I think we can implement your suggestion in the next milestone

**Pull request merged**